microsoft/testsuites/dpdk/common.py

```diff
@@ -382,3 +385,40 @@ def unsupported_os_thrower(os: Posix) -> bool:
         os,
         message=("Installer did not define dependencies for this os."),
     )
+
+
+def get_debian_backport_repo_args(os: Debian) -> List[str]:
+    if not isinstance(os, Debian):
+        return []
+    if isinstance(os, Ubuntu) and (
+        is_ubuntu_latest_or_prerelease(os) or not is_ubuntu_lts_version(os)
```

Comment:
`is_ubuntu_latest_or_prerelease` is not stable, we should avoid using the concept. Please use a method to check the distro version like 20.04, 22.04.

Reply:
Compromise: I will check the versions in that function.

Reply:
I don't understand why the latest os are not supported. Is it a temp period, or should be always true? Please add comments in the code.

Reply:
I think you're misunderstanding the core concept. The 'backports' args are only needed for old OS's (specifically debian distros). So this function decides whether to add an extra argument to `apt _____` to use the backports repository over the regular one. This is to get more recent versions of packages that are backported to work with the older OS. It's an optional flag but we want it because there are bug fixes in there.

This isn't deciding whether to support an OS or not, it's just deciding whether to add this extra backports repo argument to apt.

Reply:
The repo isn't available for latest or non-lts verisons of debian distros, also. So we don't include it in those cases. They don't need backported stuff because they're already using latest.

Reply:
I think I also included that filter for latest because occasionally there _would_ be a jammy-backports repo but it would raise an error if you tried to use it explicitly. I'll comment this function more, it's not clear enough currently.

Reply:
Commented. I don't want to mess with this more than I need to, it was already checked in and is needed for bionic/buster.

Reply:
The "latest" is not a stable concept. I understand like 20.04/22.04 needs backports to support dpdk. But it doesn't mean in 2026, the 24.04 needs backports. It may make the test cases suddenly fail after 2026/1/1.

Reply:
Well... wait. I switched this code to _not_ use a date comparison because you commented that latest wasn't a stable concept. Can I switch it back? Previously the function looked at whether the OS was over two years old (since that's the release schedule Canonical uses)

Reply:
I just check the method name, haven't check content of the method. That's great to see the date check was removed! How about use a name like `ubuntu_need_backports`? The "latest" is confusing, because 24.04 is not "latest" at 2026, but it doesn't need the backports.

Reply:
I kind of want to revert this to what is was before. It's unrelated to the refactor. I get that we always can improve the code but... this was already reviewed and approved. Let's change it in it's own PR if we're going to change it.

Reply:
> The "latest" is not a stable concept. I understand like 20.04/22.04 needs backports to support dpdk. But it doesn't mean in 2026, the 24.04 needs backports. It may make the test cases suddenly fail after 2026/1/1.
> 
> So, the logic should be like `if version <= 22.04, then install new kernels`.

No, this _isn't_ right. There are unsupported interrum releases of ubuntu where we don't want to try to use the backport repo. 21.11 or 21.04 don't have backport repos because they're not supported past the release of 22.04.

This is why this is written this way, checking if it's LTS or latest. This also _isn't_ the function where the kernel is updated. It is just to get this specific backport argument.

Reply:
undoing my previous change. The check was fine and working before

Reply:
> No, this _isn't_ right. There are unsupported interrum releases of ubuntu where we don't want to try to use the backport repo. 21.11 or 21.04 don't have backport repos because they're not supported past the release of 22.04.

I'm trying to provide logic about LTS versions, you can still have other logic to check if it's LTS or not. But for LTS versions, it shouldn't rely on actual years. It's highly like to be failed in 2026.

Reply:
> I just check the method name, haven't check content of the method. That's great to see the date check was removed! How about use a name like `ubuntu_need_backports`? The "latest" is confusing, because 24.04 is not "latest" at 2026, but it doesn't need the backports.

Ah ok this makes a lot of sense. Will change the name and comment things.

I need to disagree about using the date though. That check is exactly to get around having a hard-coded 'latest' that we have to update every two years. Until Canonical changes their release schedule and decides to release an LTS version on an odd year or move to a rolling release or something, the function will continue to work as intended. We need to enable the repo for releases which are LTS versions that are older than two years (the initial support window). This function does that.

see https://ubuntu.com/about/release-cycle

It's been like this as far back as I can find and there aren't plans to change it as far as I can tell.

I've restructured the code and added more thorough comments to explain things. Let's finish this review up?

Reply:
Let me try to discuss from a question. Today, we know the 24.04 doesn't need backports to work well. How can we know it needs backports, and there are backports for it at 2026/1/1?